src/mesh/smoothing.rs

```diff
@@ -320,12 +318,9 @@ pub fn loop_subdivision(
                     mid_vertex_indices[edge_index] = Some(mid_vertex_index);
                 }
 
-                let mid_v1v2_index =
-                    mid_vertex_indices[0].expect("Must have been produced by earlier loop");
-                let mid_v2v3_index =
-                    mid_vertex_indices[1].expect("Must have been produced by earlier loop");
-                let mid_v3v1_index =
-                    mid_vertex_indices[2].expect("Must have been produced by earlier loop");
+                let mid_v1v2_index = mid_vertex_indices[0]?;
+                let mid_v2v3_index = mid_vertex_indices[1]?;
+                let mid_v3v1_index = mid_vertex_indices[2]?;
```

Comment:
Did this really crash here or you just weren't sure? From what I remember this really shouldn't happen, unlike the other condition that you fixed.